conf/config.go

```diff
@@ -190,3 +216,102 @@ func UnsetEnv(protect *testing.T, key string) error {
 	return err
 
 }
+
+/***********************************************************************************************
+
+    CODE BELOW THIS BLOCK IS EXPERIMENTAL AND ONLY BEING USED INTERNALLY BY CONF PACKAGE
+
+***********************************************************************************************/
+
+// Gopath function exposes the gopath of the application while keeping it immutable. Golang does
+// not allow const to be strings, and a var would make it mutable if made public. This could be
+// something the config / viper struct keeps track. For now it's separate.
+func Gopath() string {
+	return envVars.gopath
+}
+
+/*
+Checkout function takes a reference to a struct or a slice of string. It will traverse both
+data structures and look up key value pairs in the conf / viper struct by the name of the field
+for structs and string values for the slice. The function works with pointers so no value is
+returned. An error is returned if the wrong data structure is provided.
+*/
+func Checkout(list interface{}) error {
+
+	// Check if the data type provided is supported
+	switch list := list.(type) {
+	// If it's a slice of strings
+	case []string:
+
+		for n, str := range list {
+			if val, exists := LookupEnv(str); exists {
```

Comment:
I left this alone just in case the user may have a "default" value in there if conf package is not tracking. But if you think "default" value should not a be a thing, it makes sense to replace it with an empty string.

Reply:
I might be missing the implementation that you have in mind.

if I have the following config values that I want to retrieve:
["DB_URL", "TIMEOUT", "NUM_CONNECTIONS"] and none of them exist in the config, I would be left with ["DB_URL", "TIMEOUT", "NUM_CONNECTIONS"] right?

I'm thinking it would be clearer to return ["", "", ""] to indicate that none of them are set.

Reply:
Sorry. I thought it was aimed at the struct. Yes, this makes sense!